src/DataFrame/IO/Parquet.hs

```diff
@@ -240,7 +240,9 @@ processColumnPages (maxDef, maxRep) pages pType _ maybeTypeLength = do
                     ERLE_DICTIONARY -> decodeDictV1 dictValsM maxDef defLvls nPresent afterLvls
                     EPLAIN_DICTIONARY -> decodeDictV1 dictValsM maxDef defLvls nPresent afterLvls
                     other -> error ("Unsupported v2 encoding: " ++ show other)
-
+            DictionaryPageHeader{} -> undefined
```

Comment:
There's this rule that I don't remember explicitly enforcing here but you can only have one dictionary page then the rest of the subsequent pages should be data pages. So seeing a dictionary page at this stage should be an error of sorts. 

https://parquet.apache.org/docs/file-format/data-pages/encodings/